scripts/deleteElasticRecords.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from model import Work, ESWork
+from managers import DBManager, ElasticsearchManager
+from elasticsearch_dsl import Search
+from elasticsearch.exceptions import NotFoundError, ConflictError
+from main import loadEnvFile
+
+
+def main():
+
+    '''Deleting ESC works that don't appear in the Postgresql database as works based on uuid'''
+
+    loadEnvFile('local-qa', fileString='config/{}.yaml')
+
+    dbManager = DBManager(
+        user= os.environ.get('POSTGRES_USER', None),
+        pswd= os.environ.get('POSTGRES_PSWD', None),
+        host= os.environ.get('POSTGRES_HOST', None),
+        port= os.environ.get('POSTGRES_PORT', None),
+        db= os.environ.get('POSTGRES_NAME', None)
+    )
+
+    esManager = ElasticsearchManager()
+    esManager.createElasticConnection()
+
+    dbManager.generateEngine()
+
+    dbManager.createSession()
+
+    batchSize = 200
+    searchES = Search(index=os.environ['ELASTICSEARCH_INDEX'])
+
+    for work in searchES.query('match_all').scan():    
+        if dbManager.session.query(Work) \
+            .group_by(Work.uuid) \
+            .filter(Work.uuid == work.uuid) \
+            .yield_per(batchSize) == None:
```

Comment:
Right approach! but since `work.uuid` is only going to yield one value this will only ever return one work.

You need to add an accumulator above this. Something like:
```
workUUIDs.append(work.uuid) # Initialize the array outside of the loop here
if len(workUUIDs) >= batchSize:
    # run the database query with
    uuidTuples = dbManager.session.query(Work.uuid).filter(Work.uuid == work.uuid).filter(Work.uuid._in(workUUIDs).all()
    # Upack the tuples to a flat array
    # Convert the array of ElasticSearch UUIDs and the array of postgresql UUIDs to sets()
    # Take the difference of of the ES set and the postgresql set (https://docs.python.org/3/library/stdtypes.html#frozenset.difference)
    # what you have left should be a set of UUIDs that are in ES but not in psql
    # that's what you should delete!